activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -409,10 +409,11 @@ def construct_relation_for_association_calculations
       def apply_join_dependency(relation, join_dependency)
         relation = relation.except(:includes, :eager_load, :preload).joins!(join_dependency)
 
-        if using_limitable_reflections?(join_dependency.reflections)
+        if using_limitable_reflections?(join_dependency.reflections) ||
+            relation.group_values.any? || relation.having_clause.any?
```

Comment:
`limited_ids_for` is not always safe for a GROUP BY query in the first place.
What should we do in that case?